server/routes/main.js

```diff
@@ -3,14 +3,14 @@ const router = express.Router();
 const passport = require("passport");
 
 //Main Routes - simplified for now
-
+const baseUrl = process.env.BASE_URL || 'http://localhost:3000'
```

Comment:
This and the env variable should be changed to 0AUTH_REDIRECT_URL for better clarity.